src/AggregatePublicKey.zig

```diff
@@ -0,0 +1,165 @@
+//! AggregatePublicKey definition for BLS signature scheme based on BLS12-381.
+const Self = @This();
+
+/// An aggregate public key that can be used to verify aggregate signatures.
+point: c.blst_p1 = c.blst_p1{},
+
+/// Converts an aggregate public key back to a regular public key.
+/// This converts from projective coordinates back to affine coordinates.
+pub fn toPublicKey(self: *const Self) PublicKey {
+    var pk = PublicKey{};
+    c.blst_p1_to_affine(&pk.point, &self.point);
+    return pk;
+}
+
+/// Aggregates multiple public keys into a single aggregate public key.
+/// If pks_validate is true, validates each public key before aggregation.
+///
+/// Returns an error if the slice is empty or if any public key validation fails.
+pub fn aggregate(pks: []const PublicKey, pks_validate: bool) BlstError!Self {
+    if (pks.len == 0) return BlstError.AggrTypeMismatch;
+    if (pks_validate) for (pks) |pk| try pk.validate();
+
+    var agg_pk = Self{};
+    c.blst_p1_from_affine(&agg_pk.point, &pks[0].point);
+    for (1..pks.len) |i| {
+        c.blst_p1_add_or_double_affine(&agg_pk.point, &agg_pk.point, &pks[i].point);
+    }
+    return agg_pk;
+}
+
+/// Aggregates multiple public keys using multi-scalar multiplication with randomness.
+/// This method is more efficient for large numbers of public keys and provides
+/// enhanced security through the use of randomness.
+///
+/// Errors if:
+/// - `pk` slice is empty,
+/// - `scratch` space is insufficient, or
+/// - if any public key validation fails.
+///
+/// Returns the `AggregatePublicKey` on success.
+pub fn aggregateWithRandomness(
+    pks: []*const PublicKey,
+    randomness: []const u8,
+    pks_validate: bool,
+    scratch: []u64,
+) BlstError!Self {
+    if (pks.len == 0) return BlstError.AggrTypeMismatch;
+    if (scratch.len < c.blst_p1s_mult_pippenger_scratch_sizeof(pks.len)) {
+        return BlstError.AggrTypeMismatch;
+    }
+    if (pks_validate) for (pks) |pk| try pk.validate();
+
+    var scalars_refs: [MAX_AGGREGATE_PER_JOB]*const u8 = undefined;
+    var pk_points: [MAX_AGGREGATE_PER_JOB]*const PublicKey.Point = undefined;
```

Comment:
this is the only use of PublicKey.Point (and Signature.Point is not used).
Consider not copying pointer here, using the same cast as in AggregateSignature.aggregateWithRandomness

Also consider removing Point entirely. (As an alternative, consider adding a doc comment to PublicKey, Signature, etc that says something to the effect of eg: "PublicKey is a `p1_affine`" or "PublicKey is a point in G1 in affine form." which can mitigate the possible confusion)